packages/dev/compiler/compiler.ts

```diff
@@ -74,13 +74,20 @@ export async function runCompiler(mode: 'dev' | 'build', projectDir: string = pr
         ...createEsbuildConfig(remixConfig),
         metafile: true,
         write: true,
+        color: true,
       })
       .then(async context => {
         console.log(`Building Service Worker in ${MODE} mode`);
         try {
           if (mode === 'build') {
-            return context.dispose();
+            context.watch().then(() => {
```

Comment:
we shouldn't use watch on `'build'` mode watch should only be used on dev, right?

Reply:
Apparently removing it lead to it never getting built at all in production. So I just watched it immediately disposed the watch function after

Reply:
I think this should do the trick
```suggestion
            conxtet.rebuild()
```

Reply:
Not tested yet, would definitely test this out for the next patch release. In the meantime, do you think `server()` is a better alternative to `watch()`. Not sure if that restarts the server, if it does, the watch issue is fixed in development. Would also test that out

Reply:
following [this](https://esbuild.github.io/api/#build) I think the options are
use `rebuild` instead of watch on that line or
move the mode if to the line 71 and use `build` directly on build mode

```js
if (mode === 'build') {
   esbuild
          .build({...})
          
} else {
    esbuild
          .context({
            ...
          }).then(async context => {
          try {
            console.log(`Building Service Worker in ${MODE} mode`);
            await context.watch();
            console.timeEnd(TIME_LABEL);
            console.log('Watching for changes in the service worker file...');
          } catch (error) {
          console.error(error);
        }
```

Reply:
Updated the compiler to use `rebuild()` instead